cmd/promtool/archive.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"os"
+)
+
+const FILE_PERM = 0644
+
+type archiverConfig struct {
+	archiveName string
+}
+
+type archiver interface {
+	write(fileName string, buf *bytes.Buffer) error
+	close() error
+	archive() *os.File
```

Comment:
If we use `name()` here, in [resource.go#L89](https://github.com/chyeh/prometheus/blob/PROM-1551/cmd/promtool/resource.go#L89) it will become `fmt.Println("debug complete all files written in:", rs.name())` which I believe is less descriptive. What do you think?

Reply:
Since `file` conflicts to the `tarGzFileWriter`'s `file` field, so we need to change the name of the field. Is `archive` ok? Something like:
```go
type tarGzFileWriter struct {
	tarWriter *tar.Writer
	gzWriter  *gzip.Writer
	archive   *os.File
}
```

Reply:
aah I didn't know about this, I added a PR to your branch with my suggestion also cleared few more nits from the golang linter.

Reply:
Merged! Thanks!!